cmd/argocd-application-controller/main.go

```diff
@@ -94,7 +95,8 @@ func newCommand() *cobra.Command {
 				resyncDuration,
 				time.Duration(selfHealTimeoutSeconds)*time.Second,
 				metricsPort,
-				kubectlParallelismLimit)
+				kubectlParallelismLimit,
+				ignoreStatus)
```

Comment:
I think this setting should be added to the `argocd-cm` ConfigMap, next to other settings. We already have `resource.compareoptions` ([PR](https://github.com/argoproj/argo-cd/pull/3803/files) that documents it). 

I would suggest adding `ignoreResourceStatusField` that might take one of the following value: `crd`, `all`, `off`. The `crd` is default.